File: backend/src/seed.ts
import { PrismaClient } from '@prisma/client';
import bcrypt from 'bcryptjs';

const prisma = new PrismaClient();

async function main() {
    console.log('🌱 Iniciando Seed...');

    // Criar usuário admin
    const adminPassword = await bcrypt.hash('admin123', 10);
    const admin = await prisma.user.upsert({
        where: { email: 'admin@burgerpizzahouse.com' },
        update: {},
        create: {
            name: 'Administrador',
            email: 'admin@burgerpizzahouse.com',
            password: adminPassword,
            role: 'ADMIN'
        }
    });

    console.log('✅ Admin criado:', admin.email);

    // Criar ingredientes
    await Promise.all([
        prisma.ingredient.upsert({
            where: { id: 'ing-1' },
            update: {},
            create: {
                id: 'ing-1',
                name: 'Farinha',
                unit: 'kg',
                currentStock: 50,
                minStock: 10,
                costPerUnit: 3.50
            }
        }),
        prisma.ingredient.upsert({
            where: { id: 'ing-2' },
            update: {},
            create: {
                id: 'ing-2',
                name: 'Queijo Mussarela',
                unit: 'kg',
                currentStock: 30,
                minStock: 8,
                costPerUnit: 25.00
            }
        }),
        prisma.ingredient.upsert({
            where: { id: 'ing-3' },
            update: {},
            create: {
                id: 'ing-3',
                name: 'Molho de Tomate',
                unit: 'L',
                currentStock: 20,
                minStock: 5,
                costPerUnit: 8.00
            }
        }),
        prisma.ingredient.upsert({
            where: { id: 'ing-4' },
            update: {},
            create: {
                id: 'ing-4',
                name: 'Carne Moída',
                unit: 'kg',
                currentStock: 15,
                minStock: 5,
                costPerUnit: 32.00
            }
        })
    ]);

    console.log('✅ Ingredientes criados');

    // Criar produtos
    await Promise.all([
        prisma.product.upsert({
            where: { id: 'prod-1' },
            update: {},
            create: {
                id: 'prod-1',
                name: 'Pizza Margherita',
                description: 'Molho de tomate, mussarela, manjericão',
                price: 45.90,
                cost: 15.00,
                category: 'PIZZA',
                preparationTime: 20,
                ingredients: {
                    create: [
                        { ingredientId: 'ing-1', quantity: 0.5, unit: 'kg' },
                        { ingredientId: 'ing-2', quantity: 0.3, unit: 'kg' },
                        { ingredientId: 'ing-3', quantity: 0.2, unit: 'L' }
                    ]
                }
            }
        }),
        prisma.product.upsert({
            where: { id: 'prod-2' },
            update: {},
            create: {
                id: 'prod-2',
                name: 'Pizza Calabresa',
                description: 'Molho de tomate, mussarela, calabresa, cebola',
                price: 49.90,
                cost: 18.00,
                category: 'PIZZA',
                preparationTime: 20
            }
        }),
        prisma.product.upsert({
            where: { id: 'prod-3' },
            update: {},
            create: {
                id: 'prod-3',
                name: 'Hambúrguer Clássico',
                description: 'Pão, carne, queijo, alface, tomate',
                price: 32.90,
                cost: 12.00,
                category: 'HAMBURGUER',
                preparationTime: 15,
                ingredients: {
                    create: [
                        { ingredientId: 'ing-2', quantity: 0.1, unit: 'kg' },
                        { ingredientId: 'ing-4', quantity: 0.2, unit: 'kg' }
                    ]
                }
            }
        })
    ]);

    console.log('✅ Produtos criados');
    console.log('🎉 Seed concluída com sucesso!');
}

main()
    .catch((e) => {
        console.error('❌ Erro no seed:', e);
        process.exit(1);
    })
    .finally(async () => {
        await prisma.$disconnect();
    });

File: backend/app/db/seed.py
"""
Popula o banco com dados iniciais (equivalente ao `catalog/management/commands/seed.py`
do backend Django, que por sua vez espelhava o `backend/src/seed.ts` original).

Uso:
    python -m app.db.seed
"""
import asyncio

from sqlalchemy import select

from app.core.security import hash_password
from app.db.session import AsyncSessionLocal
from app.models.catalog import Category, Ingredient, Product, ProductIngredient
from app.models.user import Role, User


async def get_or_create_ingredient(db, name: str, **defaults) -> Ingredient:
    result = await db.execute(select(Ingredient).where(Ingredient.name == name))
    ingredient = result.scalars().first()
    if ingredient is None:
        ingredient = Ingredient(name=name, **defaults)
        db.add(ingredient)
        await db.flush()
    else:
        for key, value in defaults.items():
            setattr(ingredient, key, value)
    return ingredient


async def get_or_create_product(db, name: str, **defaults) -> Product:
    result = await db.execute(select(Product).where(Product.name == name))
    product = result.scalars().first()
    if product is None:
        product = Product(name=name, **defaults)
        db.add(product)
        await db.flush()
    else:
        for key, value in defaults.items():
            setattr(product, key, value)
    return product


async def set_recipe(db, product: Product, entries: list[tuple[Ingredient, float, str]]) -> None:
    result = await db.execute(select(ProductIngredient).where(ProductIngredient.product_id == product.id))
    for pi in result.scalars().all():
        await db.delete(pi)
    await db.flush()
    for ingredient, quantity, unit in entries:
        db.add(ProductIngredient(product_id=product.id, ingredient_id=ingredient.id, quantity=quantity, unit=unit))


async def run() -> None:
    print("🌱 Iniciando seed...")
    async with AsyncSessionLocal() as db:
        result = await db.execute(select(User).where(User.email == "admin@burgerpizzahouse.com"))
        admin = result.scalars().first()
        if admin is None:
            admin = User(
                name="Administrador",
                email="admin@burgerpizzahouse.com",
                role=Role.ADMIN,
                is_superuser=True,
                hashed_password=hash_password("admin123"),
            )
            db.add(admin)
            await db.flush()
        print(f"✅ Admin: {admin.email} (senha: admin123)")

        farinha = await get_or_create_ingredient(
            db, "Farinha", unit="kg", current_stock=50, min_stock=10, cost_per_unit=3.50
        )
        queijo = await get_or_create_ingredient(
            db, "Queijo Mussarela", unit="kg", current_stock=30, min_stock=8, cost_per_unit=25.00
        )
        molho = await get_or_create_ingredient(
            db, "Molho de Tomate", unit="L", current_stock=20, min_stock=5, cost_per_unit=8.00
        )
        carne = await get_or_create_ingredient(
            db, "Carne Moída", unit="kg", current_stock=15, min_stock=5, cost_per_unit=32.00
        )
        print("✅ Ingredientes criados")

        margherita = await get_or_create_product(
            db, "Pizza Margherita",
            description="Molho de tomate, mussarela, manjericão",
            price=45.90, cost=15.00, category=Category.PIZZA, preparation_time=20,
        )
        await set_recipe(db, margherita, [
            (farinha, 0.5, "kg"),
            (queijo, 0.3, "kg"),
            (molho, 0.2, "L"),
        ])

        await get_or_create_product(
            db, "Pizza Calabresa",
            description="Molho de tomate, mussarela, calabresa, cebola",
            price=49.90, cost=18.00, category=Category.PIZZA, preparation_time=20,
        )

        classico = await get_or_create_product(
            db, "Hambúrguer Clássico",
            description="Pão, carne, queijo, alface, tomate",
            price=32.90, cost=12.00, category=Category.HAMBURGUER, preparation_time=15,
        )
        await set_recipe(db, classico, [
            (queijo, 0.1, "kg"),
            (carne, 0.2, "kg"),
        ])
        print("✅ Produtos criados")

        await db.commit()
        print("🎉 Seed concluída com sucesso!")


if __name__ == "__main__":
    asyncio.run(run())
